src/connector.js

```diff
@@ -0,0 +1,612 @@
+}
+
+// Listen to URL change (hash)
+window.addEventListener( 'hashchange', onHashChange );
+
+// Unsubscribe to controllers
+const unsubscribeManager = urlManager.subscribe( () => {
+	if ( !params.enableHistoryPush	|| window.location.origin.startsWith( 'file://' ) ) {
+		return;
+    }
+		
+	let hash = `#${urlManager.state.fragment}`;
+
+	if ( window.location.href.includes( '#' ) ) {
+		hash = hash.replace( '#', '&' );
+    }
+	
+	if ( !statusController.state.firstSearchExecuted ) {
+		window.history.replaceState( null, document.title, window.location.href + hash );
+		return;
+	}
+	window.history.pushState( null, document.title, window.location.href + hash );
+});
+
+let searchBoxState = null;
+let resultListState = null;
+let querySummaryState = null;
+let pagerState = null;
+
+let lastCharKeyUp = null;
+
+// Show query suggestions if a search action was not executed (if enabled)
+function updateSearchBoxState( newState ) {
+	searchBoxState = newState;
+	if ( !suggestionsElement ) {
+		return;
+    }
+		
+	if ( lastCharKeyUp == 13 ) {
+		suggestionsElement.style.display = 'none';
+		return;
+	}
+
+	if ( !searchBoxState.isLoadingSuggestions ) {
+		suggestionsElement.textContent = '';
+		searchBoxState.suggestions.forEach( ( suggestion ) => {
+			const node = document.createElement( "li" );
+			node.setAttribute( "class", "suggestion-item" );
+			node.onclick = ( e ) => { 
+                searchBoxController.selectSuggestion(e.currentTarget.innerText);
+                searchBoxElement.value = e.currentTarget.innerText;
+            };
+			node.innerHTML = suggestion.highlightedValue;
+			suggestionsElement.appendChild( node );
+		});
+		
+		if ( searchBoxState.suggestions.length > 0 ) {
+			suggestionsElement.style.display = 'block';
+		}
+	}
+}
+
+// Filters out dangerous URIs that can create XSS attacks such as `javascript:`.
+function filterProtocol( uri ) {
+
+	const isAbsolute = /^(https?|ftp|file|mailto|tel):/i.test( uri );
+	const isRelative = /^(\/|\.\/|\.\.\/)/.test( uri );
+
+	return isAbsolute || isRelative ? uri : '';
+}
+
+// Clear results list
+function clearResultList() {
+	resultListElement.textContent = "";
+}
+
+// Update results list
+function updateResultListState( newState ) {
+	resultListState = newState;
+	
+	if ( resultListState.isLoading ) {
+		if ( suggestionsElement ) {
+			suggestionsElement.style.display = 'none';
+        }
+	}
+	else {
+		clearResultList();
+		if( !resultListState.hasError && resultListState.hasResults ) {
+			resultListState.results.forEach( ( result, index ) => {
+				const sectionNode = document.createElement( "section" );
+				const highlightedExcerpt = HighlightUtils.highlightString( {
+					content: result.excerpt,
+					highlights: result.excerptHighlights,
+					openingDelimiter: '<strong>',
+					closingDelimiter: '</strong>',
+				} );
+				let itemId = `result-${index}`;
+				const resultDate = new Date( result.raw.date );
+
+				sectionNode.innerHTML = resultTemplateHTML
```

Comment:
If my understanding of the code is right, I think we will have a double section element here which would fail SC 1.3.1 info and relationship. I am going to validate that later via the test page.

Reply:
MANDATORY